src/UserSpaceInstrumentation/AddressRange.h

```diff
@@ -15,10 +18,21 @@ namespace orbit_user_space_instrumentation {
 struct AddressRange {
   AddressRange() = default;
   AddressRange(uint64_t start, uint64_t end) : start(start), end(end) {}
+
+  friend bool operator==(const AddressRange& lhs, const AddressRange& rhs) {
+    return lhs.start == rhs.start && lhs.end == rhs.end;
+  }
```

Comment:
For consistency this definition can also be moved out of the class, like in the example in the link.

Reply:
I'd like to keep the definition inline since I don't have a  .cpp file for this class and don't want to create one just for this one line. Or am I missing a way to get this done?

Reply:
Yeah, but you can just put it next to `AbslHashValue`, that's also not in a cpp file.

Reply:
Like here: https://abseil.io/docs/cpp/guides/hash#making-hashable-types

Just for consistency, otherwise I'm wondering why `AbslHashValue` is not inline while `operator==` is.

Reply:
Or maybe I am missing something?

Reply:
> Yeah, but you can just put the definition next to AbslHashValue, that's also not in a cpp file.

But then the definition gets inserted into every compilation unit that includes this header and I get an odr violation.  

> I'm wondering why `AbslHashValue `is not inline while `operator==` is.

AbslHashValue is templated so this does not apply.

Reply:
Do you think the example at the link only works in a .cpp? My guess was that since `operator==` is declared in the class (as `friend`), that makes it inline and hence no ODR violation. It's just a hypothesis but it got me curious now.

Reply:
Otherwise `operator==` outside of the class can be `inline` explicitly and again you shouldn't have ODR problems.

Reply:
> Do you think the example at the link only works in a .cpp? My guess was that since operator== is declared in the class (as friend), that makes it inline and hence no ODR violation. It's just a hypothesis but it got me curious now.

If the example is in a header (non-trivial, included multiple times) it does not work. One can either make the definition outside of the class explicitly inline (as you suggested) or move the definition into the class (which also makes it inline). 
To my understanding `friend` does not imply `inline`. I think there might be a difference between these solutions in terms of how the lookup works. 

In my case I don't even need the friend declaration since all members of AddressRange are public. So now I just defined the operator as an inline function outside the class. Turns out I also don't need the include I added before (the example / link mentioned that).